spring-cloud-stream-binder-kinesis-core/src/main/java/org/springframework/cloud/stream/binder/kinesis/properties/KinesisBinderConfigurationProperties.java

```diff
@@ -26,4 +26,13 @@
 @ConfigurationProperties(prefix = "spring.cloud.stream.kinesis.binder")
 public class KinesisBinderConfigurationProperties {
 
+	private boolean autoCreateStreams = true;
+
+	public boolean getAutoCreateStreams() {
+		return autoCreateStreams;
```

Comment:
Great, @JacobASeverson ! 

I'm glad that we with you are in the same boat.

I wasn't involved in the Kafka Binder design, so can't say what is the point to allow to have that behavior there. Maybe main point that Kafka allows to create topic on demand by its nature: http://kafka.apache.org/documentation.html#configuration. See `auto.create.topics.enable`. Therefore even if we configure there `autoCreateStreams=false` and no topic on the Broker in advance, it will be created automatically by the first interaction with it - consume or produce.

Here in Kinesis we don't have such a luxury, hence the confuse with this option.

So, we should create stream in the Kinesis if it doesn't exist unconditionally.
Although I guess we could have `updateShardCount = true/false` option and call an appropriate Kinesis method if provides shards count isn't equal to the existing in stream. Otherwise we just return the stream as is overriding provided `partitions` options.